### DISCUSSION TITLE
Gain SNR and flag estimations during quartical solution and meaing of chisq stats provided

### DISCUSSION THREAD

**Discussion by devojyoti96** [ratt-ru/QuartiCal#335] [Q&A]
Hi,
I have some questions regarding how signal-to-noise of individual gain solutions are determined and flags are computed in quartical.

1. In CASA, there is one parameter one can supply is minsnr during solving gain terms. If a gain solution has found to have SNR less than this minsnr, that solution is flagged. I am wondering how that is being done in quartical, because there is no such parameter.
2. Second point is that, if minsnr is not there, how can I estimate the uncertainty of the value it estimated. For example, let say quartical estimated phases of an antenna is 0.4 deg. Should one believe this value as true or just comning due to noise in the measurement/ numerical uncertainty. Is there any way to quantify?
3. I am asking this particularly for off-diagonal terms in complex-2x2 solutions. Sometimes these could be very small and if plotted again frequency channels or number of antennas may look like having a zero mean. But individual values are not exactly zero. Now, should we consider these off-diagonal terms are zero and just coming from measurement noise? Or is there any way to quantify the error on them?
4. Next question is how flags are computed during solution? What basis it is computed?
5. Final question is about the chisq statistics. Can you explain in detail what the post solve chisq statistics shown by different colours actually means?

Hope to hear back from you soon. 
Devojyoti

**Comment by JSKenyon** [comment]
Hi @devojyoti96! 

> 1. In CASA, there is one parameter one can supply is minsnr during solving gain terms. If a gain solution has found to have SNR less than this minsnr, that solution is flagged. I am wondering how that is being done in quartical, because there is no such parameter.

As you correctly point out, we don't do this in QuartiCal (yet). I would need to look at precisely how CASA does this estimate. 

> 2. Second point is that, if minsnr is not there, how can I estimate the uncertainty of the value it estimated. For example, let say quartical estimated phases of an antenna is 0.4 deg. Should one believe this value as true or just comning due to noise in the measurement/ numerical uncertainty. Is there any way to quantify?

QuartiCal does save the diagonal entries of J^HJ (from the GN algorithm) which provides a measure of the variance on the gains. This is still a work in progress and may require some interpretation. These estimates are also omitted for parameterised terms. 

> 3. I am asking this particularly for off-diagonal terms in complex-2x2 solutions. Sometimes these could be very small and if plotted again frequency channels or number of antennas may look like having a zero mean. But individual values are not exactly zero. Now, should we consider these off-diagonal terms are zero and just coming from measurement noise? Or is there any way to quantify the error on them?

This is difficult to answer as it will depend on the instrument. Leakage values may be small and may vary with frequency. I would need to see the plots and know a little more about the data in question to comment. 

> 4. Next question is how flags are computed during solution? What basis it is computed?

QuartiCal uses a somewhat bespoke approach which attempts to track and flag diverging solutions i.e. if a solution consistently exhibits divergent behaviour it is flagged. This is not perfect and I am open to alternative suggestions.

> 5. Final question is about the chisq statistics. Can you explain in detail what the post solve chisq statistics shown by different colours actually means?

In the output printed per data chunk, colour merely indicates whether the chisq increased or decreased in the chunk. For run-end statistics, the explanation is printed in the log:

![image](https://github.com/ratt-ru/QuartiCal/assets/6582745/4b75bb2a-cf47-4a04-95e8-8c0109597acf)
 
Here the mean is the mean chisq value over all chunks. Simply, this is a very rudimentary way of spotting chunks which contain bad data/solutions.

**Reply by devojyoti96** [reply]
Hi @JSKenyon, 
2. If quartical saves the diagonal entries for j^hj, what is the issue of not using it for determining SNR of the solutions? I think it is important to implement, because otherwise including low SNR solutions during polselfcal may cause problem? What do you think? Can one use the J^HJ inverse to estimate the errors on gain terms, if it is not implemented in quartical yet? Is there any potential timeline when we can see this in quartical? 
 
3. Here are the plots of off-diagonal terms for question 3. I know that the leakage values can be small. But, how to quantify the errors on them. 
![correlation-XY (1)](https://github.com/ratt-ru/QuartiCal/assets/52938718/3d5ad0a7-6209-411e-a4ed-5bb77e58800f)
![correlation-XY](https://github.com/ratt-ru/QuartiCal/assets/52938718/1c71d42e-3cbf-49b3-9470-828231d7ee74)
![correlation-YX (1)](https://github.com/ratt-ru/QuartiCal/assets/52938718/03e1d1f4-72a1-422f-abf7-482e68a7024e)
![correlation-YX](https://github.com/ratt-ru/QuartiCal/assets/52938718/3738101f-2240-4767-8833-269ec1c66a78)
